lib/sanitize/transformers/clean_element.rb

```diff
@@ -72,6 +98,27 @@ def call(env)
               attr.unlink
             end
           end
+        
+          # Leading and trailing whitespace around URLs is ignored at parse
+          # time. Stripping it here prevents it from being escaped by the
+          # libxml2 workaround below.
+          attr.value = attr.value.strip
```

Comment:
It seems at this point it is already possible that `attr` has been unlinked, but the `do` loop continues with the same `attr`. Should the new code somehow be made conditional on whether `attr` has been unlinked yet, for performance reasons? (No use doing the `strip` and the `gsub` if `attr` is already sanitized, I take it?)